test/integ_tests/test_device_creation.py

```diff
@@ -74,43 +75,61 @@ def test_get_devices_all():
         assert arn in result_arns
 
 
+def _get_provider_name(device: AwsDevice) -> str:
+    arn_provider = device.arn.split("/")[2]
+
+    # capitalize as in provider name
+    provider_primary_name = device.provider_name.split()[0]
+    if arn_provider == provider_primary_name.lower():
+        capitalized = provider_primary_name
+    else:
+        capitalized = arn_provider.upper()
+
+    # remove dashes
+    return capitalized.replace("-", "")
+
+
+def _get_device_name(device: AwsDevice) -> str:
+    arn_device_name = device.arn.split("/")[-1]
+
+    device_name = arn_device_name.replace("_", "").replace("-", "").replace("system", "")
```

Comment:
I'm afraid that in the future we'll have a device with "system" in it and we won't want to remove it and we'll be confused. But I don't think it's the end of the world to change this then.

Reply:
I agree, I've updated this PR to be more explicitly about this transformation (Advantage_system -> Advantage)